### DISCUSSION TITLE
Typing and scrolling feels smooth in released version than compiled version!!

### DISCUSSION THREAD

**Discussion by dineshKumar777** [neovide/neovide#738] [Q&A]
First off all thanks for this cool project. Really loving using neovide on daily basis. 
I downloaded neovide from releases and using it in my win 10 machine. 
Reason I started liking neovide right away is that smooth cursor movement typing animation like in MS office products. It felt so smooth and native. And mouse scroll too.
I compiled neovide from source and started using from yesterday. I don't know why but its not feeling smooth anymore.

after going through most of the github issues, it seems you migrated to opengl from vulkan. is that the reason or I am missing some configuration for smooth typing?

**Comment by Kethku** [comment]
I'm not sure. Can you give some more details about what feels slow to you? Is it the cursor movement (this was tweaked somewhat)? Is it the time from key press to character showing up in the screen? Could it be the framerate (we swapped to winit so its possible things are rendering slower?

Its hard to test this sort of thing out without some extra details about what exactly feels slower. Making the editor feel responsive is very high on my priority list, so I'm committed to getting it right, it just might take some back and forth to land on what exactly is going wrong.

**Reply by Kethku** [reply]
A shot in the dark: are you building it yourself? Make sure to build in release mode if  so

**Reply by dineshKumar777** [reply]
Its more like cursor movement (there is no lag in keypress at all). I will try to record a gif or video later to show the diff.
I don't have experience with RUST, so not sure how to benchmark these details.

I built as mentioned in Readme (cargo build --release).

**Reply by IBS27** [reply]
I tried to install the m1 dmg build from the actions page. But when I run it and open the app I get a message saying that app is damaged. My system is an m1 mac mini and its running macos 11.4. Please help me! I really want to try out neovide. I even tried building it but it didn't work.

**Reply by dineshKumar777** [reply]
@IBS27 I am a full time windows user.
It seems this issue already addressed in https://github.com/Kethku/neovide/issues/409
Check there. 
Otherwise try neovide from releases section. Its old, but i believe it will work fine in M1. Give it a try. I am using this version in my machine. https://github.com/Kethku/neovide/releases/tag/0.5.0

**Reply by IBS27** [reply]
I tried all that. I have been trying this for the past week. Nothing seems to work. 
@Kethku I wanted to know when the homebrew cask feature will roll out. That may fix my problems.

**Reply by IBS27** [reply]
I am actually an avid neovim user. But I tried the smooth caret and scrolling in vscode (and loved it!). So, I am using neovim embedded in vscode. If I can get neovide working, I won't need to do that.

**Reply by dineshKumar777** [reply]
yeah smooth caret in windows products are awesome. I love to type in my outlook and msword. vscode experience is also close to that but little less. I was amazed to get the same experience, when I tried neovide in releases section first time. Because in windows typing in terminal is not as smooth as in linux.
But sadly upon compiling from source I am not getting that smooth caret. that's why created this discussion. 
@Kethku I tried screen recording, but i don't think you wont able to notice. If you have windows machine, try typing in neovide from following release 
https://github.com/Kethku/neovide/releases/tag/0.5.0
and compare typing with latest source! Lets see how it goes.

**Reply by IBS27** [reply]
There are no good tutorials on youtube for building the binaries. I think @Kethku should create a youtube channel and upload tutorials and stuff for neovide. It would be very useful.

**Reply by IBS27** [reply]
And btw in the releases what is that first file for?

**Reply by dineshKumar777** [reply]
@IBS27  notsure

**Reply by Kethku** [reply]
Thats the linux build

**Reply by IBS27** [reply]
Oh ok

**Comment by dineshKumar777** [comment]
@Kethku I tried screen recording, but I don't think you will be able to notice. See the cursor movement in both attachments. If you have windows machine, try typing in neovide from following release
https://github.com/Kethku/neovide/releases/tag/0.5.0
and compare typing with latest source! Lets see how it goes.
Ref video: https://www.youtube.com/watch?v=Yc5cHSZ_9b4&t=11s

https://user-images.githubusercontent.com/21097695/122934442-289a9180-d38d-11eb-9148-ac449ae97167.mp4


https://user-images.githubusercontent.com/21097695/122934454-2a645500-d38d-11eb-83cf-9a1044cb9009.mp4

**Reply by Kethku** [reply]
Interesting. So sounds like this is a a framerate issue. I have some ideas

**Reply by dineshKumar777** [reply]
Glad you can able to see the issue here. let me know if you need help with testing this. Thanks
